Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2485,7 +2485,7 @@ extension BrowserViewController: ContextMenuHelperDelegate {
 
         if let url = elements.image {
             if dialogTitle == nil {
-                dialogTitle = url.absoluteString
+                dialogTitle = elements.title ?? url.absoluteString.ellipsize(maxLength: ActionSheetTitleMaxLength)
```

Comment:
I moved the `ellipsize` method here (from line 2556) so that only URLs are "ellipsized". Titles will always be shown in full, which matches Safari's behavior.

Reply:
From a quick test, looks like the URL may be up to around 3 lines long.
![](https://user-images.githubusercontent.com/19424103/55576022-df8b9380-56d5-11e9-95f7-80b0805c5a17.png)